ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineGetBlobsV1.java

```diff
@@ -0,0 +1,155 @@
+ * `B`, the response **MUST** be `[A, null, C]`.
+ *
+ * <p>3. Client software **MUST** support request sizes of at least 128 blob versioned hashes. The
+ * client **MUST** return `-38004: Too large request` error if the number of requested blobs is too
+ * large.
+ *
+ * <p>4. Client software **MAY** return an array of all `null` entries if syncing or otherwise
+ * unable to serve blob pool data.
+ *
+ * <p>5. Callers **MUST** consider that execution layer clients may prune old blobs from their pool,
+ * and will respond with `null` if a blob has been pruned.
+ */
+public class EngineGetBlobsV1 extends ExecutionEngineJsonRpcMethod {
+
+  private final Map<VersionedHash, BlobsWithCommitments.BlobQuad> blobMap = new HashMap<>();
+  private final BlobCache blobCache;
+
+  public EngineGetBlobsV1(
+      final Vertx vertx,
+      final ProtocolContext protocolContext,
+      final EngineCallListener engineCallListener,
+      final TransactionPool transactionPool) {
+    super(vertx, protocolContext, engineCallListener);
+    this.blobCache = transactionPool.getBlobCache();
+    transactionPool.subscribePendingTransactions(this::onTransactionAdded);
+    transactionPool.subscribeDroppedTransactions(this::onTransactionDropped);
+  }
+
+  @Override
+  public String getName() {
+    return "engine_getBlobsV1";
+  }
+
+  @Override
+  public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext) {
+    final VersionedHash[] versionedHashes;
+    try {
+      versionedHashes = requestContext.getRequiredParameter(0, VersionedHash[].class);
+    } catch (JsonRpcParameter.JsonRpcParameterException e) {
+      throw new InvalidJsonRpcParameters(
+          "Invalid versioned hashes parameter (index 0)",
+          RpcErrorType.INVALID_VERSIONED_HASHES_PARAMS,
+          e);
+    }
+
+    if (versionedHashes.length > 128) {
+      return new JsonRpcErrorResponse(
+          requestContext.getRequest().getId(),
+          RpcErrorType.INVALID_ENGINE_GET_BLOBS_V1_TOO_LARGE_REQUEST);
+    }
+
+    final List<BlobAndProofV1> result = getResult(versionedHashes);
+
+    return new JsonRpcSuccessResponse(requestContext.getRequest().getId(), result);
+  }
+
+  private @Nonnull List<BlobAndProofV1> getResult(final VersionedHash[] versionedHashes) {
+    return Arrays.stream(versionedHashes)
+        .map(this::getBlobQuad)
+        .map(this::getBlobAndProofV1)
+        .toList();
+  }
+
+  private @Nullable BlobAndProofV1 getBlobAndProofV1(final BlobsWithCommitments.BlobQuad bq) {
+    if (bq == null) {
+      return null;
+    }
+    return new BlobAndProofV1(
+        bq.blob().getData().toHexString(), bq.kzgProof().getData().toHexString());
+  }
+
+  private BlobsWithCommitments.BlobQuad getBlobQuad(final VersionedHash vh) {
+
+    BlobsWithCommitments.BlobQuad blobQuad = blobMap.get(vh);
+    if (blobQuad == null) {
+      blobQuad = blobCache.get(vh);
+    }
```

Comment:
in theory this fallback should not be necessary, and removing the need to have the `blobCache` will simplify a lot the change, removing the need to pass it around, or am I missing something?

Reply:
I also don't understand. It seems to me that blobMap is redundant, and this RPC should just use the same instance of the blobCache that the transactionpool does. Management of the cache contents should be ignored by the RPC and treated as a "read only" concern.

Reply:
The blob cache does contain all the blobs that have been put into a blob recently. These blobs are used to re-add the blob transactions in case of a reorg, as the blobs are not part of the block. After a 3 epochs these blobs are removed from the cache.
The blob map keeps track of all the blobs that are part of transactions that are in our transaction pool. These are the blobs that the CL will most likely ask for. We do keep them in the map for as long as their transactions are in the transaction pool. 
@fab-10 @jflo

Reply:
According to the specification of the new method, the CL should be interested in blobs that are in the txpool, and not blobs that have been already included in a block, so `blobMap` should be enough, because in case of a reorg a new add notification will be sent.